modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -102,6 +110,10 @@ class PublishDir {
 
     private String taskName
 
+    TaskProcessor taskProcessor
+
+    TaskRun taskRun
+
```

Comment:
Still not convinced about adding those fields just to access the session since it's introducing an unnecessary dependency with task context. 

It's true that *now* `publishDir` can only be executed within a process, however a plan is to make it also available as an operator or from with in a workflow (some discussion [here](https://github.com/nextflow-io/nextflow/discussions/1933)), and in that case there would be not `TaskRun` and `TaskProcessor` instances. 

Think it would be better just using the `Global.session` object, which I agree it's ugly, but it's already used within this class. 

It also would be interesting planning the use of some minimal DI framework to replacing the the use of `Global` object (but this is another story..)

Reply:
Yes, I agree about unnecessary dependencies ( also I evaluated to define a `PublishDirListener` or similar but doesn't like to much)

Ok, I'll use the `Global`